plugin/storage/factory_config.go

```diff
@@ -22,9 +22,12 @@ import (
 )
 
 const (
-	// SpanStorageTypeEnvVar is the name of the env var that defines the type of backend used for span storage.
+	// SpanStorageTypeEnvVar is the name of the env var that defines the type of backend used for span writing.
 	SpanStorageTypeEnvVar = "SPAN_STORAGE_TYPE"
 
+	// SpanReaderTypeEnvVar is the optional env var that defines the type of backend used for span reading in the case of multi-storage.
+	SpanReaderTypeEnvVar = "SPAN_READER_TYPE"
```

Comment:
have you tried without introducing this new config? What's not working without it?

Reply:
Since we can't have a CompositeReader, in the case of multiple storage types we must decide on 1 storage to read from. At the moment, the first listed storage type is used and a warning is emitted, but the extra flag allows users to specify the storage so that they don't have to be concerned with the order of their inputted storage types.

Reply:
> but the extra flag allows users to specify the storage so that they don't have to be concerned with the order of their inputted storage types.

I suggest removing this optional behavior - the same can be achieved by reordering items in the main storage type var without additional code.

Reply:
> the same can be achieved by reordering items in the main storage type var without additional code.

I'm not a fan of this non user friendly behavior where the user has to know the ordering of the flags, the new variable makes it explicit.

Reply:
gonna be a pedant and not forward this conversation in anyway; the ingester will read from kafka and write to kafka/cassandra, this won't be just a marginal use case. As a user, I'd prefer a second explicit flag.